log10/bigquery.py

```diff
@@ -0,0 +1,67 @@
+from google.cloud import bigquery
+from google.api_core.exceptions import NotFound
+import os
+
+
+# todo: add requirements.txt file
+# todo: add instructions for bigquery integration
+
+def initialize_bigquery():
+
+    # Configure the BigQuery client
+    project_id = os.environ.get("PROJECT_ID")
+    dataset_id = os.environ.get("DATASET_ID")
+    table_id = os.environ.get("TABLE_ID")
+
+    client = bigquery.Client(project=project_id)
+
+    def dataset_exists(dataset_id):
+        try:
+            client.get_dataset(dataset_id)  # API request
+            return True
+        except NotFound:
+            return False
+
+    def table_exists(dataset_id, table_id):
+        try:
+            table_ref = client.dataset(dataset_id).table(table_id)
+            client.get_table(table_ref)  # API request
+            return True
+        except NotFound:
+            return False
+
+    # Check if dataset exists
+    if dataset_exists(dataset_id):
+        print(f"Dataset {dataset_id} exists.")
+    else:
+        print(f"Dataset {dataset_id} does not exist. Creating...")
+        # Create the dataset
+        dataset_ref = client.dataset(dataset_id)
+        dataset = bigquery.Dataset(dataset_ref)
+        # Set the location, e.g., "US", "EU", "asia-northeast1", etc.
+        dataset.location = "US"
```

Comment:
Maybe we should have this from an environment, too. Can be done later if we want to add it to the backlog for now